File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        // count frequency
        for(int num : nums){
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        //Find majority element
        int n = nums.length;
        for (int key : map.keySet()) {
            if (map.get(key) > n / 2) {
                return key;
            }
        }
        return -1;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        for ele in nums:
            freq[ele] = freq.get(ele, 0) + 1

        for key,value in freq.items():
            if value > len(nums) // 2:
                return key

        return
        

        